### PR TITLE
Add command to configure callhome options

### PR DESCRIPTION
**Pull request by anjalshireesh**
mc support callhome set {alias} logs=on|off

This will internally create/enable/disable a logger webhook called `subnet`
The `auth_token` of the webhook will be set to the SUBNET api key,
which will be fetched from the minio config. In case the cluster is not yet
registered, an error will be raised asking the user to register it first.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -46,6 +46,7 @@ require (
 require (
 	github.com/charmbracelet/bubbles v0.10.0
 	github.com/charmbracelet/lipgloss v0.4.0
+	github.com/minio/minio-go v6.0.14+incompatible
```
Comment (vadmeste) [inline go.mod:49]: Can you do `go mod tidy` ?

Comment (harshavardhana) [reply go.mod:49]: you need to depend on `github.com/minio/minio-go/v7`  most probably you are depending on `github.com/minio/minio-go` in one of your imports @anjalshireesh

File: cmd/support-callhome-set.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/probe"
+	"github.com/minio/minio-go/pkg/set"
```
Comment (harshavardhana) [inline cmd/support-callhome-set.go:26]:
```suggestion
	"github.com/minio/minio-go/v7/pkg/set"
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/support-callhome-set.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+func getOptionKV(input string) (string, string, error) {
+	so := supportedOptions()
+	parts := strings.Split(input, "=")
+	opt := parts[0]
+	if !so.Contains(opt) {
+		return "", "", fmt.Errorf("Invalid option %s. Valid options are %s", opt, so.String())
+	}
+
+	val := parts[1]
```
Comment (vadmeste) [inline cmd/support-callhome-set.go:151]: This can crash if the user provide only 'logs'

**Review by vadmeste — COMMENTED** [review]
Few comments

File: cmd/support-callhome-set.go [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+	Before:       setGlobalsFromContext,
+	Flags:        callhomeFlags,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} TARGET option=on|off
+
+OPTIONS:
+  logs - Push minio log entries to SUBNET
```
[the remark this answers is not in the document]
Comment (anjalshireesh) [reply cmd/support-callhome-set.go:49]: It seems realtime during my testing, but couldn't find it being mentioned in documentation of the logger webhook. So I'm not sure.

File: cmd/support-callhome-set.go [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+  2. Disable logs callhome for cluster with alias 'play'
+     {{.Prompt}} {{.HelpName}} play logs=off
+`,
+}
+
+func supportedOptions() set.StringSet {
+	return set.CreateStringSet("logs")
+}
+
+func validVals() set.StringSet {
```
[the remark this answers is not in the document]
Comment (anjalshireesh) [reply cmd/support-callhome-set.go:67]: These are not specific to logs, but will be used for switching on/off any of the callhome options (to be added in future). So renaming to `validOptionValues`

File: cmd/support-callhome.go [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+var supportCallhomeCmd = cli.Command{
+	Name:         "callhome",
+	Usage:        "Configure the callhome functionality",
+	OnUsageError: onUsageError,
+	Action:       mainCallhome,
+	Before:       setGlobalsFromContext,
+	Flags:        globalFlags,
+	Subcommands:  callhomeSubcommands,
+	CustomHelpTemplate: `NAME:
```
[the remark this answers is not in the document]
Comment (anjalshireesh) [reply cmd/support-callhome.go]: Removed.

File: cmd/support-callhome-set.go [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+
+	for _, option := range ctx.Args().Tail() {
+		// options have already been validated, so third (error) return value can be ignored
+		k, v, _ := getOptionKV(option)
+		if k == "logs" {
+			if v == "on" {
+				configureSubnetWebhook(aliasedURL, true)
+			} else {
+				configureSubnetWebhook(aliasedURL, false)
+			}
```
[the remark this answers is not in the document]
Comment (anjalshireesh) [reply cmd/support-callhome-set.go]: Thanks, done.

File: cmd/support-callhome-set.go (hunk 3 above)
Comment (anjalshireesh) [reply cmd/support-callhome-set.go:154]: Thanks, fixed.

File: cmd/support-callhome-set.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (vadmeste) [reply cmd/support-callhome-set.go]: > It seems realtime during my testing, but couldn't find it being mentioned in documentation of the logger webhook. So I'm not sure.

It is realtime. I just saw we are enabling a simple http webhook

[the remark this answers is not in the document]
Comment (anjalshireesh) [reply cmd/support-callhome-set.go]: ok, changed the text to include 'realtime'

File: cmd/support-callhome-set.go [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+	// Get the alias parameter from cli
+	alias, _ := url2Alias(aliasedURL)
+
+	// Create a new MinIO Admin Client
+	client, err := newAdminClient(aliasedURL)
+	fatalIf(err, "Unable to initialize admin connection.")
+
+	apiKey := getSubnetAPIKeyFromConfig(alias)
+	if len(apiKey) == 0 {
+		err := fmt.Errorf("Please register the cluster first by running 'mc admin subnet register %s'", alias)
```
Comment (harshavardhana) [inline cmd/support-callhome-set.go:116]:
```suggestion
		e := fmt.Errorf("Please register the cluster first by running 'mc admin subnet register %s'", alias)
```

Go error is `e` and probe.Error is `err`

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**